### ISSUE TITLE
Steering not turning for the Autopilotv2 and i am using the same run video and same code

### ISSUE THREAD

**Issue by vineethamurali** [akshaybahadur21/Autopilot#3]

**Comment by abhinavsagar** [comment]
@vinzi2810 How did you manage to run autopilotV2 as there are no such files in this repo?

**Comment by maxmstrmn** [comment]
Same here. Have you solved your problem?

**Comment by Sriram97** [comment]
Same here.

**Comment by maxmstrmn** [comment]
Did you succeed in the meantime?

**Comment by Sriram97** [comment]
Yes ita working now.

**Comment by maxmstrmn** [comment]
What did you do? Could you share the source code?

**Comment by Sriram97** [comment]
Hi,

I didn't change anything. We just need to delete the label and features
file from the folder and run it again. Load the file again, train it and run the AutopilotV2 file.

**Comment by maxmstrmn** [comment]
Hi, thank you for this hint. But it doesn't work. I only use the v2 files from this repo (which don't include labels), train and test. The final file size of the h5 file is around 8mb, which is wrong as the file shipped in this repo ist 28mb large.
Also if I print the network summary of both files it seems that some layers are missing in my file...
Which environment do you use?

**Comment by Sriram97** [comment]
HI,

Yes it does not have labels and features. You must run load data.py first.
Then it will create labels and features. For that, you must download a
dataset from sully chen and store it inside the Autopilotv2 folder. Then
run train.py. The model will be trained and stored in .h file. After that,
run the AutopilotAppv2.py. It will work

**Comment by maxmstrmn** [comment]
Hi,
I ran Load_DataV2.py on the dataset with 45000 images. This creates the labels and features files (~2GB size). Then I run the training script which creates the .h5 file, but this file is not as large as the file in this repo. 
I keep trying using the larger dataset, but there must be something wrong.
I use Windows 10, Python 3.6, Keras 3.2.1 Tf 2.0.0 GPU.

**Comment by maxmstrmn** [comment]
No success, even with the larger dataset the final .h5 file is around 8mb large.

**Comment by maxmstrmn** [comment]
It seems the training is crazy. I printed both model summaries using the print_summary method from the keras utils. As you can see, in my trained model, a lot of layers are missing!

Here ist the output:

summary .h5 from this repo
Model: "sequential_1"
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
=================================================================
lambda_1 (Lambda)            (None, 100, 100, 1)       0         
_________________________________________________________________
conv2d_1 (Conv2D)            (None, 100, 100, 32)      320       
_________________________________________________________________
activation_1 (Activation)    (None, 100, 100, 32)      0         
_________________________________________________________________
max_pooling2d_1 (MaxPooling2 (None, 50, 50, 32)        0         
_________________________________________________________________
conv2d_2 (Conv2D)            (None, 50, 50, 64)        18496     
_________________________________________________________________
activation_2 (Activation)    (None, 50, 50, 64)        0         
_________________________________________________________________
max_pooling2d_2 (MaxPooling2 (None, 25, 25, 64)        0         
_________________________________________________________________
conv2d_3 (Conv2D)            (None, 25, 25, 128)       73856     
_________________________________________________________________
activation_3 (Activation)    (None, 25, 25, 128)       0         
_________________________________________________________________
max_pooling2d_3 (MaxPooling2 (None, 12, 12, 128)       0         
_________________________________________________________________
flatten_1 (Flatten)          (None, 18432)             0         
_________________________________________________________________
dropout_1 (Dropout)          (None, 18432)             0         
_________________________________________________________________
dense_1 (Dense)              (None, 128)               2359424   
_________________________________________________________________
dense_2 (Dense)              (None, 64)                8256      
_________________________________________________________________
dense_3 (Dense)              (None, 1)                 65        
=================================================================
Total params: 2,460,417
Trainable params: 2,460,417
Non-trainable params: 0
_________________________________________________________________


summary .h5 from my training output

Model: "sequential_1"
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
=================================================================
lambda_1 (Lambda)            (None, 100, 100, 1)       0         
_________________________________________________________________
conv2d_1 (Conv2D)            (None, 100, 100, 32)      320       
_________________________________________________________________
activation_1 (Activation)    (None, 100, 100, 32)      0         
_________________________________________________________________
max_pooling2d_1 (MaxPooling2 (None, 50, 50, 32)        0         
_________________________________________________________________
conv2d_2 (Conv2D)            (None, 50, 50, 32)        9248      
_________________________________________________________________
activation_2 (Activation)    (None, 50, 50, 32)        0         
_________________________________________________________________
max_pooling2d_2 (MaxPooling2 (None, 25, 25, 32)        0         
_________________________________________________________________
conv2d_3 (Conv2D)            (None, 25, 25, 64)        18496     
_________________________________________________________________
activation_3 (Activation)    (None, 25, 25, 64)        0         
_________________________________________________________________
max_pooling2d_3 (MaxPooling2 (None, 12, 12, 64)        0         
_________________________________________________________________
conv2d_4 (Conv2D)            (None, 12, 12, 64)        36928     
_________________________________________________________________
activation_4 (Activation)    (None, 12, 12, 64)        0         
_________________________________________________________________
max_pooling2d_4 (MaxPooling2 (None, 6, 6, 64)          0         
_________________________________________________________________
conv2d_5 (Conv2D)            (None, 6, 6, 128)         73856     
_________________________________________________________________
activation_5 (Activation)    (None, 6, 6, 128)         0         
_________________________________________________________________
max_pooling2d_5 (MaxPooling2 (None, 3, 3, 128)         0         
_________________________________________________________________
conv2d_6 (Conv2D)            (None, 3, 3, 128)         147584    
_________________________________________________________________
activation_6 (Activation)    (None, 3, 3, 128)         0         
_________________________________________________________________
max_pooling2d_6 (MaxPooling2 (None, 1, 1, 128)         0         
_________________________________________________________________
flatten_1 (Flatten)          (None, 128)               0         
_________________________________________________________________
dropout_1 (Dropout)          (None, 128)               0         
_________________________________________________________________
dense_1 (Dense)              (None, 1024)              132096    
_________________________________________________________________
dense_2 (Dense)              (None, 256)               262400    
_________________________________________________________________
dense_3 (Dense)              (None, 64)                16448     
_________________________________________________________________
dense_4 (Dense)              (None, 1)                 65        
=================================================================
Total params: 697,441
Trainable params: 697,441
Non-trainable params: 0
_________________________________________________________________

**Comment by Sriram97** [comment]
Did u try to run the AutopilotV2.py? Is the steering angle turning or still
static?

**Comment by maxmstrmn** [comment]
Everything is tested with the v2 files. The angle is still static. Please note the remarkable difference between these two models I posted above.

If I use the model .h5 shipped in this repo everything works fine. But my own trained fails.

**Comment by Sriram97** [comment]
Yes true. I just realised the new model trained is only 8.258 mb whereas
the real model is 28.89 mb. Something wrong somewhere

**Comment by Sriram97** [comment]
However, I have tested with own dataset and the steering angle is moving.
It's weird but working

**Comment by Shagun23** [comment]
hello i am getting an error while running code

**Comment by hasithram** [comment]
could you please share the whole code for AutopilotV2.py.
Because when i went through the whole code there was no difference between V1 and V2 code(the Autopilot V1.py and Autopilot V1.py)
And even there is no *"run.mp4"* video for V2
Could you share these stuff...would be really helpful

**Comment by akshaybahadur21** [comment]
The run.mp4 for v2 is not uploaded due to copyright issues.
Use video from youtube
